src/keyboard/mod.rs

```diff
@@ -0,0 +1,258 @@
+//! Utilities for keyboard handling
+//!
+//! This module provides utilities for keyboardand keymap handling: keymap interpretation
+//! and forwarding keystrokes to clients using xkbcommon.
+//!
+//! You can first create a `KbdHandle` using the `create_keyboard_handler` function in this module.
+//! The handle you obtained can be cloned to access this keyboard state from different places. It is
+//! expected that such a context is created for each keyboard the compositor has access to.
```

Comment:
I see... I guess it would be worth having a look at what WLC and Weston do to match libinput with the wayland seats then.

Anyway, this PR explicitly stops at the boundary of "one logical keyboard", be it a single one or multiple ones. So I guess we could move this discussion to a dedicated issue, as I think this is not blocking here?